server/command.go

```diff
@@ -131,6 +139,39 @@ func (c *Conn) dispatch(data []byte) interface{} {
 		}
 
 		return eofResponse{}
+	case COM_REGISTER_SLAVE:
+		if h, ok := c.h.(ReplicationHandler); ok {
+			if err := h.HandleRegisterSlave(data); err != nil {
+				return err
+			}
+			return nil
+		} else {
+			return fmt.Errorf("the handler does not support replication protocol, use ReplicationHandler instead")
+		}
+	case COM_BINLOG_DUMP:
+		if h, ok := c.h.(ReplicationHandler); ok {
+			pos, _ := parseBinlogDump(data)
+			s := replication.NewBinlogStreamer()
+			go h.HandleBinlogDump(pos, s)
+
+			return s
```

Comment:
another choice is that this Conn never returns, we directly call writeBinlogEvents in a loop. I'm not sure what should we do if the Conn is returned and MySQL client send another command to this connection

Reply:
Interesting question. mysql-clients can be disconnected when they are not responsive (see [net_write_timeout](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_net_write_timeout) and [net_read_timeout](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_net_read_timeout)). We should check how replication protocol works.

Reply:
Thanks for reminding us, there're two questions

1. I wrongly thought the connection will be re-used for next command at first, and want to remind author to check what's the behaviour if the connection is both serving a binlog stream and client's request. Now I see `writeBinlogEvents` won't do that.
2. mysql client may [set heartbeat interval](https://github.com/go-mysql-org/go-mysql/blob/81966e15b9c5b1c66847a52c6f194771885ef64e/replication/binlogsyncer.go#L324-L325) and binlog streamer should append heartbeat event when the stream is idle for the interval. I think it's OK to implement it later because some use case of this feature will not let stream be idle for a long time.